src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3355,6 +3365,15 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, Number
                                 return true;
                             }
 
+                            result = null;
+                            return false;
+                        case NumberSuffixFlags.SignedByte:
+                            if (sbyte.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out sbyte sb))
+                            {
+                                result = (sbyte)(sb * multiplier);
```

Comment:
Seems we could return false if multiplier > 1 and SignedByte or UnsignedByte. I do not know whether it makes sense to do this test explicitly. Certainly this will eliminate the exception.

Reply:
Well, still needs to be parsed as (s)byte regardless, no?

But yes, taking the shortcut of skipping that if there is a multiplier applied here is a code call, thank you!

Reply:
>  still needs to be parsed as (s)byte regardless

I think yes. It allow "recovery" and continue parsing.

Reply:
Aaaand patched in, thank you!

Reply:
And again, good catch.

The multiplier can be *negative* in valid signed parsing cases because the hex parsing ignores negative sign, deferring that to the multiplier to be applied after parsing of the hex literal. I assume this is because the hex parsing methods of `.TryParse()` do not allow negative signs, which is probably because hex parsing permits entering signed values by utilising the sign bit.

Reply:
It is not obvious behavior so please add the comment in code.

Reply:
Iexpect `result = sb;`

Reply:
Impossible to completely ignore multiplier; if a hex literal is negative, multiplier will be `-1` by this point thanks to code referenced above. 

We must still either multiply by multiplier value or else account for it in a more complex expression, e.g., `result = multiplier < 0 ? -sb : sb`

I think simply multiplying by the already-verified multiplier value is the most straightforward method.

Reply:
I agrre with this in the PR but I'd consider the possible optimization in follow PR.

Reply:
Yeah, this will be resolved more cleanly in the follow up PR. 😄